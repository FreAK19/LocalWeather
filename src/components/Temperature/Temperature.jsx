import React from 'react';
import PropTypes from 'prop-types';
import './Temperature.less'

const Temperature = ({ temp }) => (
  <div className='weather__temp'>
    <p>{temp}</p>
  </div>
);

Temperature.propTypes = { temp: PropTypes.number.isRequired };

export default Temperature;
