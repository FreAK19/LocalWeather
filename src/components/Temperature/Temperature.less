
.weather__temp {
  font-weight: 900;
  font-size: 5rem;
  text-align: center;
  flex-basis: 50%;
  align-self: center;

  & > p {
    margin: 0;
  }
}