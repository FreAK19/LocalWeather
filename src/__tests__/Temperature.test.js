import React from 'react';
import Temperature from '../components/Temperature/Temperature';

test('Temperature should render correctly', () => {
  const component = shallow(<Temperature temp={45} />);
  expect(component).toMatchSnapshot();
});

test('render correct temperature base on props', () => {
  const component = shallow(<Temperature />);
  component.setProps({ temp: 40 });
  expect(component.find('p').text()).toMatch('40');
});

test('should round temperature if it is float number', () => {
  const component = shallow(<Temperature temp={56.453} />);
  expect(component.find('p').text()).toMatch('56');
});
