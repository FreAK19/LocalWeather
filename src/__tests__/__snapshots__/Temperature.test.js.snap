// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature should render correctly 1`] = `
<div
  className="weather__temp"
>
  <p>
    45
  </p>
  <span>
    C
  </span>
</div>
`;
